components/utils.ts

```diff
@@ -123,8 +123,77 @@ export function extractToken(
     next?.("router")
 }
 
+export const DEFAULT_MASTERY_MAXLEVEL = 20
+export const XP_PER_LEVEL = 6000
+
+export function getMaxProfileLevel(gameVersion: GameVersion): number {
+    if (gameVersion === "h3") {
+        return 7500
+    }
+
+    return 5000
+}
+
+/**
+ * Calculates the level for the given XP based on XP_PER_LEVEL.
+ * Minimum level returned is 1.
+ */
+export function levelForXp(xp: number): number {
+    return Math.max(1, Math.floor(xp / XP_PER_LEVEL) + 1)
+}
+
+/**
+ * Calculates the required XP for the given level based on XP_PER_LEVEL.
+ * Minimum XP returned is 0.
+ */
 export function xpRequiredForLevel(level: number): number {
-    return level * 6000 - 6000
+    return Math.max(0, (level - 1) * XP_PER_LEVEL)
+}
+
+//TODO: Determine some mathematical function
```

Comment:
I'll do this, the difference between levels "steadily" increases until it doesn't:
![image](https://user-images.githubusercontent.com/24512050/226173019-3b71c9c2-8f13-44e6-a2fb-8b27771eca9c.png)